src/semgrep_agent/meta.py

```diff
@@ -325,16 +325,31 @@ def ci_job_url(self) -> Optional[str]:
 
     @cachedproperty
     def event_name(self) -> str:
-        return os.getenv("CI_PIPELINE_SOURCE", "unknown")
+        gitlab_event_name = os.getenv("CI_PIPELINE_SOURCE", "unknown")
+        if gitlab_event_name in ["merge_request_event", "external_pull_request_event"]:
+            return "pull_request"
+        return gitlab_event_name
 
     @cachedproperty
     def pr_id(self) -> Optional[str]:
         return os.getenv("CI_MERGE_REQUEST_IID")
 
+    @cachedproperty
+    def start_sha(self) -> Optional[str]:
+        return os.getenv("CI_MERGE_REQUEST_DIFF_BASE_SHA")
+
     @cachedproperty
     def pr_title(self) -> Optional[str]:
         return os.getenv("CI_MERGE_REQUEST_TITLE")
 
+    def to_dict(self) -> Dict[str, Any]:
+        return {
+            **super().to_dict(),
+            "branch": self.commit_ref,
+            "base_sha": self.base_commit_ref,
+            "start_sha": self.start_sha,
```

Comment:
adds two environment variables that are necessary for posting inline MR comments in GitLab